src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,244 @@
+		String inputLower = input.toLowerCase();
+		if (inputLower.isBlank()) {
+			return;
+		}
+		String[] inputArr = inputLower.split("\\|");
+		TaskType operationType = inputLower.startsWith("t")
+				? TaskType.TODO
+				: inputLower.startsWith("d")
+						? TaskType.DEADLINE
+						: TaskType.EVENT;
+
+		Task newTask = null;
+
+		switch (operationType) {
+			case TODO:
+				newTask = new Todo(inputArr[2]);
+				break;
+
+			case DEADLINE:
+				newTask = new Deadline(inputArr[2], parseDateTimeStr(inputArr[3], ui, storage, dialogContainer));
+				break;
+
+			case EVENT:
+				newTask = new Event(inputArr[2],
+						new LocalDateTime[] { parseDateTimeStr(inputArr[3], ui, storage, dialogContainer),
+								parseDateTimeStr(inputArr[4], ui, storage, dialogContainer) });
+				break;
+
+			default:
+				break;
+		}
+		if (inputArr[1].equals("1"))
+			newTask.markAsDone();
+		tasklist.add(newTask);
+	}
+
+	/**
+	 * Convert String datetime to LocalDateTime object.
+	 * 
+	 * @param dateTimeStr
+	 * @param ui
+	 * @param storage
+	 * @param dialogContainer
+	 * @return LocalDateTime
+	 */
+	public LocalDateTime parseDateTimeStr(String dateTimeStr, Ui ui, Storage storage, VBox dialogContainer) {
+		if (dateTimeStr.isBlank()) {
+			return null;
+		}
+		String[] dateTimeArr = dateTimeStr.split(" ");
+		LocalTime time = LocalTime.MIN;
+		Integer[] dateArr;
+		String errorMessage = "Oops! Datetime format needs to be specified in proper form.";
+		if (dateTimeArr.length > 1) {
+			try {
+				int hr = Integer.parseInt(dateTimeArr[1].substring(0, 2));
+				int min = Integer.parseInt(dateTimeArr[1].substring(2, 4));
+				time = LocalTime.of(hr, min);
+			} catch (NumberFormatException ex) {
+				ui.sendResponse(dialogContainer, storage, ui.createLabel(errorMessage));
+				return null;
+			}
+		}
+
+		try {
+			if (dateTimeStr.contains("/")) { // Input Format: dd/MM/YYYY
+				dateArr = Stream.of(dateTimeArr[0].split("/"))
+						.map(Integer::valueOf)
+						.toArray(Integer[]::new);
+				return LocalDateTime.of(LocalDate.of(dateArr[2], dateArr[1], dateArr[0]), time);
+			}
+
+			// Input Format: YYYY-MM-dd
+			dateArr = Stream.of(dateTimeArr[0].split("-"))
+					.map(Integer::valueOf)
+					.toArray(Integer[]::new);
+			return LocalDateTime.of(LocalDate.of(dateArr[0], dateArr[1], dateArr[2]), time);
+		} catch (ArrayIndexOutOfBoundsException ex) {
+			ui.sendResponse(dialogContainer, storage, ui.createLabel(errorMessage));
+			return null;
+		}
+
+	}
+
+	/**
+	 * Get a datetime object from deadline tasks.
+	 * 
+	 * @param inputArr
+	 * @param ui
+	 * @param storage
+	 * @param dialogContainer
+	 * @return LocalDateTime
+	 */
+	public LocalDateTime getBy(String[] inputArr, Ui ui, Storage storage, VBox dialogContainer) {
+		int l = inputArr.length;
+		for (int i = 0; i < l; i++) {
+			if (i < l - 1 && inputArr[i].equals("/by")) {
+				return parseDateTimeStr(sliceArrAndConcate(inputArr, i + 1, l), ui, storage, dialogContainer);
+			}
+		}

```

Comment:
Perhaps use a more meaningful variable name instead of l? Since it is in a larger scope than variable i, I believe it would be better to use something else like `arrLength`